### DISCUSSION TITLE
🐞 `WithFile` fudges ownership/permissions of intermediate paths

### DISCUSSION THREAD

**Discussion by vito** [dagger/dagger#12466] [Help]
### What is the issue?

It looks like `WithFile("/a/b/c", ...)` currently changes ownership and/or permissions of `/a/b` even if they already exist.

### Steps to reproduce

h/t @stobias123: https://gist.github.com/stobias123/46c7b9dce0506349ddeb6597674f19cd

### SDK version

Go SDK v0.8.4

### OS version

Linux

**Comment by marcosnils** [comment]
also, this doesn't happen with `Dockerfile` standard `COPY`. My assumption is that this could be related to the MergeOp optimizations we have for `WithX` operations.

**Comment by mhemken-vts** [comment]
I think I'm running into an issue with this as well.
I create a file in one container and add to a different container so I can `flux push artifact`. I get a permission denied error.

```
█ [0.15s] exec id
┃ uid=65534(nobody) gid=65534(nobody) groups=65534(nobody)

█ [0.13s] exec ls -ls config.json
┃      4 -rw-r--r--    2 root     root           716 Apr  3 22:09 config.json	fmt.Println(result)
```

And I did not need linux to tell me I'm a nobody. I just want to use `config.json` for something.

**Comment by marcosnils** [comment]
hey there! one workaround for this is to supply the `permissions` argument to the `WithFile` operation so it gets set to some valid values.
